plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -364,13 +364,27 @@ func (p *PrometheusClient) addMetricFamily(point telegraf.Metric, sample *Sample
 	addSample(fam, sample, sampleID)
 }
 
+// Sorted returns a copy of the metrics in time ascending order.  A copy is
+// made to avoid modifying the input metric slice since doing so is not
+// allowed.
+func sorted(metrics []telegraf.Metric) []telegraf.Metric {
+	batch := make([]telegraf.Metric, 0, len(metrics))
+	for i := len(metrics) - 1; i >= 0; i-- {
```

Comment:
Why not just do a `copy(batch, metrics)` here, as it's faster?

Reply:
This is just a copy/paste from stackdriver output, it is reversing the order, but we are sorting anyway so it probably doesn't matter.  Could it be faster to get the buffer in approximate order before sorting?

Reply:
Ah yes, that is true.